src/main/kotlin/cacophony/codegen/instructions/GenerateAsm.kt

```diff
@@ -1,13 +1,22 @@
 package cacophony.codegen.instructions
 
+import cacophony.codegen.BlockLabel
+import cacophony.codegen.linearization.BasicBlock
+import cacophony.codegen.linearization.LoweredCFGFragment
 import cacophony.codegen.registers.RegisterAllocation
 
-fun generateAsm(instruction: Instruction, registerAllocation: RegisterAllocation): String {
-    return instruction.toAsm(registerAllocation.successful)
-}
+fun generateAsm(block: BasicBlock, registerAllocation: RegisterAllocation): String =
+    ".${block.label().name}:\n" +
+        block
+            .instructions()
+            .filterNot { it.isNoop(registerAllocation.successful) }
+            .map { it.toAsm(registerAllocation.successful) }
+            .filter(String::isNotEmpty)
```

Comment:
I think we should have a single way to test if instruction is a noop - either via isNoop, or by testing if the returned string is empty - but not both.

Reply:
I am not sure what the semantics of an empty instruction are (tbh I'm not sure if it's even possible), I just left this to not introduce possibly breaking changes